### ISSUE TITLE
Allow new nodes to be added during runtime

### ISSUE THREAD

**Issue by p-offtermatt** [informalsystems/CometMock#70]

**Comment by Anmol1696** [comment]
Ahh. And does cometmock write anything back to the validator node home? If it is just passed in memory, and not used again, maybe it might make sense to provide it directly.

**Comment by p-offtermatt** [comment]
It does not write anything there, no, so I think in principle it should be fine to pass it directly. I'd only be concerned that it becomes annoying to start CometMock, my experience with passing jsons on the command line is that it's annoying since you sometimes have to worry about escaping quotes in the right places.
could also think about allowing providing them either as files or on the cli, but that may be unnecessary complexity

**Comment by p-offtermatt** [comment]
Yes, having the node home or the two files directly seems like a great option, much better than my thought of passing the file content in directly - it makes it so normal users don't need to worry about the content of the node home, and advanced users can easily use it for other use cases as well.

Would be very happy about a PR for this if you feel up to it. The v0.38 upgrade is almost done and I will open a PR in an hour or two. If you want, you could branch off of `https://github.com/informalsystems/CometMock/tree/ph/v0.38.x-upgrade` to start working on this,

**Comment by mmulji-ic** [comment]
> With something like this we might be able to solve both our usecases, as well as this seems to be the way for configuring the current cosmos apps as well, which i actually like.

@Anmol1696 this is fine we just want to add one new validator, but we may want to add multiple at different times.

**Comment by p-offtermatt** [comment]
> > With something like this we might be able to solve both our usecases, as well as this seems to be the way for configuring the current cosmos apps as well, which i actually like.
> 
> @Anmol1696 this is fine we just want to add one new validator, but we may want to add multiple at different times.

Could you clarify what you think is the problem with what Anmol proposed?
By calling it multiple times and calling advance_time, it seems possible to add multiple validators at different times, assuming the command can be executed while the main cometmock binary is running (e.g. by calling into an rpc endpoint cometmock offers)

**Comment by mmulji-ic** [comment]
The following api ...

`cometMock --priv-val-key <path to val key> --priv-val-state <path to val state>`

looked like a one-shot execution, but if cometmock calls using the above, connect to a running instance that would work.

**Comment by p-offtermatt** [comment]
👍 you are right! I think this assumes the command is a replacement for an RPC call. There is a separate issue to provide nice One-Shot commands that hit the various endpoints cometmock offers that is helpful for context https://github.com/informalsystems/CometMock/issues/53
